src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type Item = {
+  [key: string]: string | number;
+};
+
+export function groupByKey(items: Item[], key: string): GroupsMap<Item> {
+  const sortedObject: GroupsMap<Item> = {};
+
+  items.forEach((item: Item) => {
+    const property: string | number = item[key];
+
+    if (!(property in sortedObject)) {
```

Comment:
also you can check like this
```suggestion
    if (!sortedObject[property]) {
```